File: generator.py
import sqlite3, HTMLgen

db = sqlite3.connect ("test.db")
c = db.cursor ()

list_doc = HTMLgen.SimpleDocument (title="List of posts")

post_list = []
c.execute ("select * from posts")
for row in c:
    id = row[0]
    title = row[2]
    content = row[3]

    post_doc = HTMLgen.SimpleDocument (title=title)
    post_doc.append (HTMLgen.Heading (1, title))
    post_doc.append (HTMLgen.Paragraph (content))
    post_doc_name = "post"+str(id)+".html"
    post_doc.write (post_doc_name)

    post_list.append (HTMLgen.A (post_doc_name, title))

list_doc.append (HTMLgen.BulletList (post_list))
list_doc.write ('list.html')

File: generator.php
<?php
/**
 * Simple static html generator
 */
class Generator {
	protected $data;
	protected $templates;

	public function __construct($data, $templates)
	{
		$this->data = $data;
		$this->templates = $templates;
	}

	public function generate()
	{
		$generated = '';

		foreach ($this->data as $row)
		{
			$generated .= '<br />'.PHP_EOL;
			$generated .= $this->parse($row, $this->templates['inner']).PHP_EOL;
		}

		$content = array('content' => $generated);

		return $this->parse($content, $this->templates['main']);
	}

	public function parse($data, $template)
	{
		foreach ($data as $key => $value)
		{
			$template = str_replace($this->key($key), $value, $template);
		}

		return $template;
	}

	public function key($key)
	{
		return '%%'.$key.'%%';
	}
}

class Test_Generator extends Generator {
	public function __construct()
	{
		$database = dirname(__FILE__).'/test.db';

		$db = new PDO('sqlite:'.$database);

		$result = $db->query('SELECT id, title, contents FROM posts')->fetchAll(PDO::FETCH_ASSOC);

		$this->data = $result;

		$this->templates['main'] = file_get_contents('main.tpl');
		$this->templates['inner'] = file_get_contents('inner.tpl');
	}
}

$test = new Test_Generator;
echo $test->generate();
